test/loader.test.js

```diff
@@ -47,47 +45,89 @@ const execute = async (options, source, contractName) => {
   expect(contract.abi.length).toBeGreaterThan(0);
 };
 
-beforeEach(() => {
-  jest.resetAllMocks();
-});
-
-test('Runs truffle compile, zos push, and zos update commands to produce fresh .json files', async (done) => {
-  const contractName = 'Contract';
-  const { network } = defaultOptions;
-  await execute(defaultOptions, contractFilePath, contractName);
-  expect(util.exec).toHaveBeenCalledTimes(2);
-  expect(util.exec).toHaveBeenCalledWith(`zos update ${contractName} --network ${network}`, execOptions);
-  expect(util.exec).toHaveBeenCalledWith(`zos push --network ${network}`, execOptions);
-  done();
-});
-
-test('Runs truffle compile, zos push, and zos update commands to produce fresh .json files with contract not being the same as filename', async (done) => {
-  const { network } = defaultOptions;
-  const contractName = 'Counter';
-  await execute(defaultOptions, `${contractFilePath}?contract=${contractName}`, contractName);
-  expect(util.exec).toHaveBeenCalledTimes(2);
-  expect(util.exec).toHaveBeenCalledWith(`zos update ${contractName} --network ${network}`, execOptions);
-  expect(util.exec).toHaveBeenCalledWith(`zos push --network ${network}`, execOptions);
-  done();
-});
-
-test('Serves json files from file system while disabled', async (done) => {
-  const contractName = 'Contract';
-  await execute(disabledOptions, contractFilePath, contractName);
-  expect(util.exec).toHaveBeenCalledTimes(0);
-  done();
-});
-
-test('Discovers parent contracts as dependencies', async (done) => {
-  const ret = ['B.sol', 'Base.sol'];
-  const deps = await getLocalDependencies('C', path.resolve(contractsBuildDir), path.resolve(contractsDir));
-  expect(deps.length).toEqual(2);
-  expect(deps.map(dep => path.basename(dep))).toEqual(ret);
-  done();
-});
-
-test('Discovers parent contracts as dependencies for contract inside one .sol file with many contracts', async (done) => {
-  const deps = await getLocalDependencies('Contract', path.resolve(contractsBuildDir), path.resolve(contractsDir));
-  expect(deps.length).toEqual(0);
-  done();
+describe('Hot Loader', () => {
+  beforeAll(() => {
+    util.exec = jest.fn();
+    util.packageExist = jest.fn();
+    util.which.sync = jest.fn();
+  });
+
+  beforeEach(() => {
+    jest.clearAllMocks();
```

Comment:
How was the experience playing with jest? I'm thinking about a possible `zos test` in the future, and whether we want to base it on mocha or jest. Now that I think of it, you had already done some research while here in Buenos Aires, right?

Reply:
I think Jest is much easy to setup and start compare to mocha & sinon & whatever combination.  Not having the burden of picking libraries is liberating. I vote for Jest for the sake of a quick bootstrap and a developer on-boarding.